contracts/strategy/LiquidationStrategy.sol

```diff
@@ -230,180 +309,158 @@ contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCall
-                }
-                _depositStablecoin(amountNeededToPayDebt, _vars.liquidatorAddress);
-                emit LogFlashLiquidationSuccess(
-                    _vars.liquidatorAddress,
-                    _debtValueToRepay,
-                    _collateralAmountToLiquidate,
-                    dexAmountOut,
-                    fathomStablecoinReceived,
-                    path
-                );
-            } else {
-                // Condition #2 if there is loss, don't sell on DEX
-                require(fathomStablecoin.balanceOf(address(this)) >= amountNeededToPayDebt, "flashLendingCall: not enough to repay debt");
-                _depositStablecoin(amountNeededToPayDebt, _vars.liquidatorAddress);
-                idleWXDC.WXDCAmount += retrievedCollateralAmount;
-                idleWXDC.amountNeededToPayDebt += amountNeededToPayDebt;
-                idleWXDC.averagePriceOfWXDC = idleWXDC.amountNeededToPayDebt.mul(WAD).div(idleWXDC.WXDCAmount);
-                emit LogFlashLiquidationSuccess(
-                    _vars.liquidatorAddress,
-                    _debtValueToRepay,
-                    _collateralAmountToLiquidate,
-                    dexAmountOut,
-                    amountNeededToPayDebt,
-                    path
-                );
+        // Retrieve collateral token from CollateralTokenAdapter
+        uint256 retrievedCollateralAmount = _retrieveCollateral(_vars.tokenAdapter.collateralToken(), _vars.tokenAdapter, _collateralAmountToLiquidate);
+        // +1 to compensate for precision loss with division
+        uint256 amountNeededToPayDebt = _debtValueToRepay.div(RAY) + 1;
+        uint256 fathomStablecoinReceivedV2;
+        uint256 fathomStablecoinReceivedV3;
+        if (uniswapV3Info[_vars.routerV3].universalRouter == address(0) && _vars.routerV2 != address(0)) {
+            _vars.v2Ratio = TENK; // Adjust to use V2 fully
+        }
+        // If bot tells the strategy to not use DEX
+        if (_vars.routerV2 == address(0) && _vars.routerV3 == address(0)) {
+            if (fathomStablecoin.balanceOf(address(this)) < amountNeededToPayDebt) {
+                revert NotEnoughToRepayDebt();
             }
-        } else {
-            // Condition #3 loss is allowed, so if there is loss, make this address pay for the loss.
-            uint256 minAmountOut = dexAmountOut < amountNeededToPayDebt ? dexAmountOut : amountNeededToPayDebt;
-            //if allowLoss is true, then the LiquidatorStrategy will pay the difference of amountNeededToPayDebt and dexAmountOut
-            uint256 fathomStablecoinReceived = _sellCollateral(
-                _vars.tokenAdapter.collateralToken(),
-                path,
-                _vars.router,
+            _depositStablecoin(amountNeededToPayDebt, _vars.liquidatorAddress);
+            idleCollateral[_vars.tokenAdapter.collateralToken()].CollateralAmount += retrievedCollateralAmount;
```

Comment:
in this function we call `_vars.tokenAdapter.collateralToken()` multiple times, but we can call once and assign the value to a variable